packages/cli/templates/react/src/components/FieldPluginExample/index.spec.tsx

```diff
@@ -1,43 +1,166 @@
+import {
+  isAssetModalChangeMessage,
+  isGetContextMessage,
+  isHeightChangeMessage,
+  isModalChangeMessage,
+  isPluginLoadedMessage,
+  isValueChangeMessage,
+} from '@storyblok/field-plugin'
 import { render, screen } from '@testing-library/react'
+import userEvent from '@testing-library/user-event'
 import { describe, test, expect, vi } from 'vitest'
 
 import FieldPlugin from '.'
-import { useFieldPlugin } from '@storyblok/field-plugin/react'
-import { FieldPluginResponse } from '@storyblok/field-plugin'
-
-vi.mock('@storyblok/field-plugin/react')
-
-const fieldPluginDefault: FieldPluginResponse<number | undefined> = {
-  type: 'loaded',
-  data: {
-    isModalOpen: false,
-    content: undefined,
-    options: {},
-    spaceId: undefined,
-    storyLang: '',
+
+const fakeContainer = (listener) => {
```

Comment:
It's not properly typed, and we should revisit all the namings. Once we improve this, we could possibly reuse this logic back into the Container. (the basic implementation of receiving what event and sending what event are the same).

Reply:
what about 'mockContainer' ?